src/js/helper/calc.js

```diff
@@ -0,0 +1,43 @@
+export const getValueList = (domList) => {
+  const arrayList = Array.from(domList)
+  return arrayList.map((target) => Number(target.value))
+}
+
+const getMatchingCnt = (enteredNumberList, winningLotto, bonusNumber) => {
+  const matchingArray = enteredNumberList.filter((num1) => winningLotto.some((num2) => num1 === num2))
+  const isBonus = winningLotto.some((number) => number === bonusNumber)
+  return { matchCnt: matchingArray.length, isBonus }
+}
+
+export const getWinningNumberCnt = (enteredNumberList, winningLottoList, bonusNumber) => {
+  const winningCnt = {
+    three: 0,
+    four: 0,
+    five: 0,
+    fiveBonus: 0,
+    six: 0,
+  }
+
+  winningLottoList.forEach((list) => {
+    const matchInfo = getMatchingCnt(enteredNumberList, list, bonusNumber)
+
+    if (matchInfo.matchCnt === 3) return (winningCnt.three += 1)
+    if (matchInfo.matchCnt === 4) return (winningCnt.four += 1)
+    if (matchInfo.matchCnt === 5 && matchInfo.isBonus) return (winningCnt.fiveBonus += 1)
+    if (matchInfo.matchCnt === 5) return (winningCnt.five += 1)
+    if (matchInfo.matchCnt === 6) return (winningCnt.six += 1)
+  })
```

Comment:
```suggestion
  const defaultWinningCounts = {
    three: 0,
    four: 0,
    five: 0,
    fiveBonus: 0,
    six: 0,
  }
  
  return lottos
    .map(lotto => getMatchCount(enteredNumberList, lotto, bonusNumber)
    .reduce((winningCounts, { matchCount, isMatchedBonus }) => { 
      switch (matchCount) {
        case 3:
          return { ...winningCounts, three: winningCounts + 1 }
         ...
         case 5:
           return isMatchedBonus 
                        ? { ...winningCounts,  fiveBonus: winningCounts.fiveBonus + 1 } 
                        : {...winningCounts, five: winningCounts.five + 1 } ; 
           ...
       }    
     }, defaultWinningCounts)
```

forEach를 map과 reduce로 분리해서, 사이드이펙트없이 각자 한 가지 처리 동작만 맡도록 개선해볼 수 있을 것 같아요.